src/main/java/com/igot/cb/discussion/service/impl/AnswerPostReplyServiceImpl.java

```diff
@@ -175,15 +176,20 @@ public ApiResponse createAnswerPostReply(JsonNode answerPostDataReplyData, Strin
                 String createdBy = answerPostReplyDataNode.get(Constants.CREATED_BY).asText();
                 String firstName = helperMethodService.fetchUserFirstName(createdBy);
                 log.info("Notification trigger started for create answerPost");
-                if(!userId.equals(discussionOwner)) {
+                if (!userId.equals(discussionOwner)) {
                     notificationTriggerService.triggerNotification(REPLIED_COMMENT, ENGAGEMENT, List.of(discussionOwner), TITLE, firstName, notificationData);
                 }
                 if (CollectionUtils.isNotEmpty(userIdList)) {
-                    Map<String, Object> replyNotificationData = Map.of(
-                            Constants.COMMUNITY_ID, answerPostReplyDataNode.get(Constants.COMMUNITY_ID).asText(),
-                            Constants.DISCUSSION_ID, jsonNodeEntity.getDiscussionId()
-                    );
-                    notificationTriggerService.triggerNotification(TAGGED_COMMENT, ENGAGEMENT, userIdList, TITLE, firstName, replyNotificationData);
+                    List<String> filteredUserIdList = userIdList.stream().filter(uniqueId -> !uniqueId.equals(discussionOwner)).collect(Collectors.toList());
```

Comment:
should be 
userIdList.stream().filter(uniqueId -> !uniqueId.equals(discussionOwner)).toList();